CMakeLists.txt

```diff
@@ -1,133 +1,107 @@
 # BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
 
-cmake_minimum_required(VERSION 3.10...3.17)
+cmake_minimum_required(VERSION 3.14...3.18)
```

Comment:
Some of the existing settings ([`BUILD_RPATH_USE_ORIGIN`](https://cmake.org/cmake/help/latest/prop_tgt/BUILD_RPATH_USE_ORIGIN.html) require CMake 3.14.

Reply:
I think RPath stuff was added by @veprbl. I certainly have no objection to making the CMake minimal version correct, and had no reason to target CMake 3.10, unless that was a constraint imposed by manylinux. (I doubt it, as the first step in the Linux deployment is to get CMake from pip.)